notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+
+A couple things to note: 
+
+<img src="../../notebooks-images/linear_projection5.png" width="500" />
+
+- $\hat{y} + e = \vec{y}$
+- $\vec{e}$ is perpendicular to $\vec{x}$ and $\vec{1}$
+- $\hat{y} = X \hat{\theta^*}$ is the vector closest to $y$ in the vector space spanned by $\vec{x}$ and $\vec{1}$
+
+Thus, we arrive at the equation:
+
+$$X  \hat{\theta^*} + \vec{e} = \vec{y}$$
+
+Left-multiplying both sides by $X^T$:
+
+$$X^T X  \hat{\theta^*} + X^T \vec{e} = X^T \vec{y}$$
+
+Since $\vec{e}$ is perpendicular to the columns of $X$, $X^T \vec{e}$ is a column vector of $0$'s. Thus, we arrive at the Normal Equation:
+
+$$X^T X  \hat{\theta^*} = X^T \vec{y}$$
+
+From here, we can easily solve for $\hat{\theta^*}$ by left-multiplying both sides by $(X^T X)^{-1}$:
+
+$$\hat{\theta^*} = (X^T X)^{-1} X^T \vec{y}$$
+
+## Finishing up the Case Study
+
+Let's return to our case study and apply what we've learned, and explain why our solution is sound
+
+$$
+\vec{y} = \begin{bmatrix} 2 \\ 1 \\ -2  \end{bmatrix} \qquad X = \begin{bmatrix} 1 & 3 \\ 1 & 0 \\ 1 & -1 \end{bmatrix}
+$$
+
+
+$$
+\begin{align}
+\hat{\theta^*} 
+&= 
+\left(
+\begin{bmatrix} 1 & 1 & 1 \\ 3 & 0 & -1 \end{bmatrix}
+\begin{bmatrix} 1 & 3 \\ 1 & 0 \\ 1 & -1 \end{bmatrix}
+\right)^{-1}
+\begin{bmatrix} 1 & 1 & 1 \\ 3 & 0 & -1 \end{bmatrix}
+\begin{bmatrix} 2 \\ 1 \\ -2  \end{bmatrix} \\
+&= 
+\left(
+\begin{bmatrix} 3 & 2\\ 2 & 10 \end{bmatrix}
+\right)^{-1}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{30-4}
+\begin{bmatrix} 10 & -2\\ -2 & 3 \end{bmatrix}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{26}
+\begin{bmatrix} -6 \\ 22 \end{bmatrix}\\
+&=
+\begin{bmatrix} - \frac{3}{13} \\ \frac{11}{13} \end{bmatrix}
+\end{align}
+$$
+
+We have analytically found that best model for least squares regression is $f_\hat{\theta} (x_i) = - \frac{3}{13} + \frac{11}{13} x_i$. We know that our choice of $\hat{\theta}$ is sound by the mathematical property that the projection of $\vec{y}$ onto the span of the columns of $X$ yields the closest point in the vector space to $\vec{y}$. Under linear constraints using the least squares metric, solving for $\hat{\theta}$ by taking the projection guarantees us the optimal solution.
+
+## When Variables are Linearly Dependent
+
+For every additional variable, we add one column to $X$. The span of the columns of $X$ is the linear combinations of the column vectors, so adding variables only changes the span into a higher dimensional subspaces when the added variable is linearly independent of the existing ones. Otherwise, if a column is linearly dependent, it can be expressed as a linear combination of some other column. Thus, for every linear combination it contributes to, we can replace it with a linear combination of the other columns. So we have not expanded the set of possible vectors spanned by the original columns of $X$.
```

Comment:
"so adding linearly independent variables changes the span into a higher dimensional subspace." instead of "so adding variables only changes the span into a higher dimensional subspaces when the added variable is linearly independent of the existing ones."

if a column is linearly dependent, it can be expressed as a linear combo of some other column**(s)**.

Hmmm.. I would delete "Thus, for every linear combination it contributes to, we can replace it with a linear combination of the other columns."